### PR TITLE
make the application of redundant physics update idempotent to improve action glitches

### PR DESCRIPTION
**Pull request by ZappoMan**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7ed35399e9e6405f511102c46e268374ccdd9145 available at: https://gist.github.com/f5671439e210278d73ddf23e613f3a90<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8003.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8003.dmg

**Comment by ZappoMan** [comment]
Note: This PR changes the server and the client. ALL tests need to be done with this PR build, and two clients observing each other's actions.

Test Plan:
- Smoke test 
    - in particular focus on a bunch of editing of physical things
    - in particular focus on actions and two observers
- Specific Test...
    - do the following test on production to see how the system currently behaviors... it should behave relatively poorly with lots of glitches on the observer that did not "move" the box.
    - then do this test on this PR with these clients/servers... you should see significantly less glitches. You might still see some glitches, but it should be DRAMATICALLY less.

    - on Client 1 - run the following code in the console....
```
var position = Vec3.sum(MyAvatar.position, {x: 1.0, y: 0.4, z: 0.0});
        boxId = Entities.addEntity({
            type: "Box",
            name: "dummy",
            position: position,
            dimensions: {x: 0.3, y: 0.7, z: 0.3},
            gravity: {x: 0.0, y: -3.0, z: 0.0},
            damping: 0.2,
            dynamic: true
        });

        var pointToOffsetFrom = Vec3.sum(position, {x: 0.0, y: 2.0, z: 0.0});
        Entities.addAction("offset", boxId, {pointToOffsetFrom: pointToOffsetFrom,
                                             linearDistance: 2.0,
                                             // linearTimeScale: 0.005
                                             linearTimeScale: 0.1
                                            });

```
    - on Client 2 - watch that box...
    - on client 1 you can do various things to push/move that box... you should see it swing relatively smoothly on both clients 1 and 2

File: libraries/entities/src/EntityItem.h [hunk 1]
```diff
@@ -550,6 +550,22 @@ class EntityItem : public SpatiallyNestable, public ReadWriteLockable {
...
+
+    // physics related changes from the network to suppress any duplicates and make
+    // sure redundant applications are idempotent
+    glm::vec3 _lastUpdatedPositionValue;
+    glm::quat  _lastUpdatedRotationValue;
+    glm::vec3 _lastUpdatedVelocityValue;
+    glm::vec3 _lastUpdatedAngularVelocityValue;
+    glm::vec3 _lastUpdatedAccelerationValue;
+
+    quint64 _lastUpdatedPositionTimestamp = 0;
```
Comment (jherico) [inline libraries/entities/src/EntityItem.h]: member initialization should be done with braces.  Prefer `uint64_t` to `quint64`.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1075ba1599160728f35ea40c4616873f95ebf636 available at: https://gist.github.com/f5671439e210278d73ddf23e613f3a90<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8003.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8003.dmg

**Comment by AndrewMeadows** [comment]
Philip is testing this one.

**Comment by PhilipRosedale** [comment]
Tested with two machines, ran script above, tested both machines colliding, grabbing, mouse grabbing.  

View remains smooth for observer on both.  

Also tested grabbing and throwing other objects, etc. 

QA Pass for me.

**Pull request merged**